tests/test_unit.py

```diff
@@ -35,3 +42,13 @@ def test_generate_keys_returns_bytes():
 
     res1, res2 = utils.generate_keys()
     assert isinstance(res1, bytes) and (isinstance(res2, bytes))
+
+
+def test_InstanceConnectionManager_connection_string_parsing():
```

Comment:
I would consider putting all of the `InstanceConnectionManager` tests into their own file. In general, I try to map the module layout to the testing layout. 

Also when testing, you should have one test for everything that goes right(unless there are multiple ways it can go right), and then additional tests for everything that could go wrong. 

So here you want something like `test_InstanceConnectionManager_init` and you test that it was created successfully. You also want something like `test_InstanceConnectionManager_wrong_connection_string` that verifies that the correct error is thrown when the argument in improperly formatted.